src/planning.py

```diff
@@ -308,6 +322,10 @@ def _skeleton_generator(
     # repeatedly outputting the same faulty skeletons.
     visited_skeletons: Set[Tuple[_GroundNSRT, ...]] = set()
     visited_skeletons.add(tuple(root_node.skeleton))
+    if use_visited_state_set:
+        # This set will maintain (frozen) atom sets that have been fully
+        # expanded already, and ensure that we never expand redundantly.
+        visited_atom_sets = set()
```

Comment:
shouldn't `root_node.atoms` be included?

Reply:
no because it hasn't been expanded yet